File: 001_two_sum/Solution.py
class Solution(object):
  def twoSum(self, nums, target):
    """
    :type nums: List[int]
    :type target: int
    :rtype: List[int]
    """
    s = {};
    for i in range(len(nums)):
      comp = target - nums[i];
      if comp in s:
          return [s[comp], i];
      s[nums[i]] = i;


if __name__ == '__main__':
  try:
    while (True):
      sol = Solution()
      nums = [int(x) for x in input().split()]
      target = int(input())
      ans = sol.twoSum(nums, target)
      print(ans)
  except EOFError:
    pass

File: 001_two_sum/Solution.cpp
// #include <iostream>
// #include <vector>
// #include <sstream>
#include <unordered_map>
// #include <iterator>

#include "../IO.h"

class Solution {
public:
  vector<int> twoSum(vector<int>& nums, int target) {
    unordered_map<int, int> m;
    vector<int> ret;
    for (int i = 0; i < nums.size(); i++){
      int comp = target - nums[i];
      if (m.find(comp) != m.end()){
        ret.push_back(m.find(comp)->second);
        ret.push_back(i);
        return ret;
      }
      pair<int,int> added(nums[i], i);
      m.insert(added);
    }
    return ret;
  }
};


int main (int argc, char** argv) {
  Solution sol;
  while(true){
    vector<int> ls;
    int target;
    cin >> ls;
    cin >> target;
    if (!cin)
      break;
    vector<int> ans = sol.twoSum(ls, target);
    cout << ans << endl;
  }
  return 0;
}



